chrome/database.js

```diff
@@ -0,0 +1,184 @@
+
+const database = {
+    requests: [],
+}
+
+// This is not optimal. The popup and the browser extension have different local/session storages
+// So we need to use this sync storage which is limited to 8K and which is slow... 
+// Until now I don't have a nice idea on how to solve this.
+// phew...
+
+if (window.database === undefined) {
+    window.database = database;
+}
+
+function saveStorageValue(key, value) {
+    return new Promise((resolve) => {
+        chrome.storage.sync.set({ [key]: value }, function () {
+            resolve();
+        });
+    });
+}
+
+function getStorageValue(key) {
+    return new Promise((resolve) => {
+        chrome.storage.sync.get([key], function (result) {
+            resolve(result[key]);
+        });
+    });
+}
+
+const load = async () => {
+    const str = await getStorageValue("requests");
+    if (str && str != "") {
+        window.database = JSON.parse(str);
+    }
+}
+
+const save = async () => {
+    await saveStorageValue("requests", JSON.stringify(window.database));
+}
+
+function debounce(func, timeout = 500) {
+    let timer;
+    return (...args) => {
+        clearTimeout(timer);
+        timer = setTimeout(() => { func.apply(this, args); }, timeout);
+    };
+}
+
+class DatabaseAdapter {
+
+    constructor(table) {
+        this.table = table;
+    }
+
+    persist = () => {
+        return debounce(() => {
+            save();
+        })
+    }
+
+    // This part was the reason for missing SCION domain indicators.
```

Comment:
I tested it out and works fine for me. IMO it is an acceptable solution for now even the persistance logic seems quite complex for what we are doing so far. I think first we need a clear strategy how to proceed with the extension development.

Reply:
I definitely agree to this!